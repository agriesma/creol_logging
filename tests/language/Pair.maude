load creol-interpreter .
mod PROGRAM is
  protecting CREOL-SIMULATOR .
  op state : -> State [ctor] .
  eq state = {
    < "Test" : Class | Inh: noInh, Param: noVid, Att: "x" |-> null,
      Mtds: < "run" : Method | Param: noVid, Att: "_" |-> null, Code: skip ;
              return ( emp ) >,
            < "init" : Method | Param: noVid, Att: "_" |-> null,
              Code: assign( "x" ; list(int(1) :: int(2)) ) ; return ( emp ) >,
      Ocnt: 0 > } .
endm

