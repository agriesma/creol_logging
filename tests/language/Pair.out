==========================================
reduce in PROGRAM : state .
result State: {
< "Test" : Class | Inh: noInh, Param: noVid, Att: "x" |-> null, Mtds:
  < "init" : Method | Param: noVid, Att: "_" |-> null, Code: assign("x" ; list(
    int(1) :: int(2))) ; return(emp) >, < "run" : Method | Param: noVid, Att:
    "_" |-> null, Code: skip ; return(emp) >,
  Ocnt: 0 >
}
==========================================
rewrite in PROGRAM : main(state, "Test", emp) .
result State: {
< ob("Test-0") : "Test" |
  Att: "this" |-> ob("Test-0"), "x" |-> list(int(1) :: int(2)),
  Pr: idle,
  PrQ: noProc,
  Lcnt: 2 >
 
< "Test" : Class | Inh: noInh, Param: noVid, Att: "x" |-> null, Mtds:
  < "init" : Method | Param: noVid, Att: "_" |-> null, Code: assign("x" ; list(
    int(1) :: int(2))) ; return(emp) >, < "run" : Method | Param: noVid, Att:
    "_" |-> null, Code: skip ; return(emp) >,
  Ocnt: 1 >
}
Bye.
